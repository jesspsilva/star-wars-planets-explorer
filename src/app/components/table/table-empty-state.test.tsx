import { fireEvent, render, screen } from "@testing-library/react";

import TableEmptyState from "./table-empty-state";

describe("TableEmptyState component", () => {
  const mockOnClearClick = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const renderEmptyState = () => {
    return render(<TableEmptyState onClearClick={mockOnClearClick} />);
  };

  it("renders the file icon", () => {
    renderEmptyState();

    const icon = screen.getByTestId("table-empty-state-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveAttribute("width", "80");
    expect(icon).toHaveAttribute("height", "80");
  });

  it("renders the empty state messages", () => {
    renderEmptyState();

    expect(screen.getByText("No data available.")).toBeInTheDocument();
    expect(
      screen.getByText("Try to adjust your search params or get back soon.")
    ).toBeInTheDocument();
  });

  it("renders clear button with icon", () => {
    renderEmptyState();

    const clearButton = screen.getByRole("button", { name: /clear search/i });
    expect(clearButton).toBeInTheDocument();
    expect(screen.getByTestId("table-empty-state-icon")).toBeInTheDocument();
  });

  it("calls onClearClick when clear button is clicked", () => {
    renderEmptyState();
    const clearButton = screen.getByRole("button", { name: /clear search/i });

    fireEvent.click(clearButton);

    expect(mockOnClearClick).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const { container } = renderEmptyState();
    expect(container).toMatchSnapshot();
  });
});
