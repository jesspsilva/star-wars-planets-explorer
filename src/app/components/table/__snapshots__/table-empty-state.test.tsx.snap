// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableEmptyState component matches snapshot 1`] = `
<div>
  <div
    class="sc-ivxoEo gEYzIn"
  >
    <span
      class="icon"
    >
      <svg
        data-testid="table-empty-state-icon"
        fill="none"
        height="80"
        viewBox="0 0 15 15"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 2.5C3 2.22386 3.22386 2 3.5 2H9.08579C9.21839 2 9.34557 2.05268 9.43934 2.14645L11.8536 4.56066C11.9473 4.65443 12 4.78161 12 4.91421V12.5C12 12.7761 11.7761 13 11.5 13H3.5C3.22386 13 3 12.7761 3 12.5V2.5ZM3.5 1C2.67157 1 2 1.67157 2 2.5V12.5C2 13.3284 2.67157 14 3.5 14H11.5C12.3284 14 13 13.3284 13 12.5V4.91421C13 4.51639 12.842 4.13486 12.5607 3.85355L10.1464 1.43934C9.86514 1.15804 9.48361 1 9.08579 1H3.5ZM4.5 4C4.22386 4 4 4.22386 4 4.5C4 4.77614 4.22386 5 4.5 5H7.5C7.77614 5 8 4.77614 8 4.5C8 4.22386 7.77614 4 7.5 4H4.5ZM4.5 7C4.22386 7 4 7.22386 4 7.5C4 7.77614 4.22386 8 4.5 8H10.5C10.7761 8 11 7.77614 11 7.5C11 7.22386 10.7761 7 10.5 7H4.5ZM4.5 10C4.22386 10 4 10.2239 4 10.5C4 10.7761 4.22386 11 4.5 11H10.5C10.7761 11 11 10.7761 11 10.5C11 10.2239 10.7761 10 10.5 10H4.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <p>
      No data available.
    </p>
    <p>
      Try to adjust your search params or get back soon.
    </p>
    <span
      class="clear-button"
    >
      <button
        class="sc-dntaoT etmIPW"
      >
        Clear search
        <svg
          data-testid="table-empty-state-clear-icon"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0.877075 7.49988C0.877075 3.84219 3.84222 0.877045 7.49991 0.877045C11.1576 0.877045 14.1227 3.84219 14.1227 7.49988C14.1227 11.1575 11.1576 14.1227 7.49991 14.1227C3.84222 14.1227 0.877075 11.1575 0.877075 7.49988ZM7.49991 1.82704C4.36689 1.82704 1.82708 4.36686 1.82708 7.49988C1.82708 10.6329 4.36689 13.1727 7.49991 13.1727C10.6329 13.1727 13.1727 10.6329 13.1727 7.49988C13.1727 4.36686 10.6329 1.82704 7.49991 1.82704ZM9.85358 5.14644C10.0488 5.3417 10.0488 5.65829 9.85358 5.85355L8.20713 7.49999L9.85358 9.14644C10.0488 9.3417 10.0488 9.65829 9.85358 9.85355C9.65832 10.0488 9.34173 10.0488 9.14647 9.85355L7.50002 8.2071L5.85358 9.85355C5.65832 10.0488 5.34173 10.0488 5.14647 9.85355C4.95121 9.65829 4.95121 9.3417 5.14647 9.14644L6.79292 7.49999L5.14647 5.85355C4.95121 5.65829 4.95121 5.3417 5.14647 5.14644C5.34173 4.95118 5.65832 4.95118 5.85358 5.14644L7.50002 6.79289L9.14647 5.14644C9.34173 4.95118 9.65832 4.95118 9.85358 5.14644Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;
